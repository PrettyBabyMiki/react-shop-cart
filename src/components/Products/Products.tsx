import Product from './Product';

import { IProduct } from 'models';

import * as S from './style';

interface IProps {
  products: IProduct[];
}

const Products = ({ products }: IProps) => {
  return (
    <S.Container>
      {products?.map((p) => (
        <Product product={p} key={p.sku} />
      ))}
    </S.Container>
  );
};

export default Products;
