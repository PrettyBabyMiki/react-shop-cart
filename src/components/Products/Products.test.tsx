import { renderWithThemeProvider } from 'utils/test/test-utils';
import { mockProducts } from 'utils/test/mocks';

import Products from '.';

describe('[components] - Products', () => {
  const setup = () => {
    return renderWithThemeProvider(<Products products={mockProducts} />);
  };

  test('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
