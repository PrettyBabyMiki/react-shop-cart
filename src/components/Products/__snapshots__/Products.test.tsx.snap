// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components] - Products should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-furwcr lmktzQ"
      >
        <div
          class="sc-dkPtRN cEbpqD"
          tabindex="1"
        >
          <div
            class="sc-hKwDye bDGYum"
          >
            Free shipping
          </div>
          <div
            alt="Cropped Stay Groovy off white"
            class="sc-gsDKAQ"
          />
          <p
            class="sc-eCImPb hQahQF"
          >
            Cropped Stay Groovy off white
          </p>
          <div
            class="sc-jRQBWg dlshIp"
          >
            <p
              class="sc-gKclnd jlmNbV"
            >
              <small>
                $
              </small>
              <b>
                10
              </b>
              <span>
                .90
              </span>
            </p>
            <p
              class="sc-iCfMLu kFBzSL"
            >
              <span>
                or 
                9
                 x
              </span>
              <b>
                $
                1.21
              </b>
            </p>
          </div>
          <button
            class="sc-bdvvtL brAMjl"
            tabindex="-1"
          >
            Add to cart
          </button>
        </div>
        <div
          class="sc-dkPtRN PbuBX"
          tabindex="1"
        >
          <div
            class="sc-hKwDye bDGYum"
          >
            Free shipping
          </div>
          <div
            alt="Basic Cactus White T-shirt"
            class="sc-gsDKAQ"
          />
          <p
            class="sc-eCImPb hQahQF"
          >
            Basic Cactus White T-shirt
          </p>
          <div
            class="sc-jRQBWg dlshIp"
          >
            <p
              class="sc-gKclnd jlmNbV"
            >
              <small>
                $
              </small>
              <b>
                13
              </b>
              <span>
                .25
              </span>
            </p>
            <p
              class="sc-iCfMLu kFBzSL"
            >
              <span>
                or 
                3
                 x
              </span>
              <b>
                $
                4.42
              </b>
            </p>
          </div>
          <button
            class="sc-bdvvtL brAMjl"
            tabindex="-1"
          >
            Add to cart
          </button>
        </div>
        <div
          class="sc-dkPtRN gaJVfz"
          tabindex="1"
        >
          <div
            alt="Skater Black Sweatshirt"
            class="sc-gsDKAQ"
          />
          <p
            class="sc-eCImPb hQahQF"
          >
            Skater Black Sweatshirt
          </p>
          <div
            class="sc-jRQBWg dlshIp"
          >
            <p
              class="sc-gKclnd jlmNbV"
            >
              <small>
                $
              </small>
              <b>
                25
              </b>
              <span>
                .90
              </span>
            </p>
            <p
              class="sc-iCfMLu kFBzSL"
            >
              <span>
                or 
                12
                 x
              </span>
              <b>
                $
                2.16
              </b>
            </p>
          </div>
          <button
            class="sc-bdvvtL brAMjl"
            tabindex="-1"
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-furwcr lmktzQ"
    >
      <div
        class="sc-dkPtRN cEbpqD"
        tabindex="1"
      >
        <div
          class="sc-hKwDye bDGYum"
        >
          Free shipping
        </div>
        <div
          alt="Cropped Stay Groovy off white"
          class="sc-gsDKAQ"
        />
        <p
          class="sc-eCImPb hQahQF"
        >
          Cropped Stay Groovy off white
        </p>
        <div
          class="sc-jRQBWg dlshIp"
        >
          <p
            class="sc-gKclnd jlmNbV"
          >
            <small>
              $
            </small>
            <b>
              10
            </b>
            <span>
              .90
            </span>
          </p>
          <p
            class="sc-iCfMLu kFBzSL"
          >
            <span>
              or 
              9
               x
            </span>
            <b>
              $
              1.21
            </b>
          </p>
        </div>
        <button
          class="sc-bdvvtL brAMjl"
          tabindex="-1"
        >
          Add to cart
        </button>
      </div>
      <div
        class="sc-dkPtRN PbuBX"
        tabindex="1"
      >
        <div
          class="sc-hKwDye bDGYum"
        >
          Free shipping
        </div>
        <div
          alt="Basic Cactus White T-shirt"
          class="sc-gsDKAQ"
        />
        <p
          class="sc-eCImPb hQahQF"
        >
          Basic Cactus White T-shirt
        </p>
        <div
          class="sc-jRQBWg dlshIp"
        >
          <p
            class="sc-gKclnd jlmNbV"
          >
            <small>
              $
            </small>
            <b>
              13
            </b>
            <span>
              .25
            </span>
          </p>
          <p
            class="sc-iCfMLu kFBzSL"
          >
            <span>
              or 
              3
               x
            </span>
            <b>
              $
              4.42
            </b>
          </p>
        </div>
        <button
          class="sc-bdvvtL brAMjl"
          tabindex="-1"
        >
          Add to cart
        </button>
      </div>
      <div
        class="sc-dkPtRN gaJVfz"
        tabindex="1"
      >
        <div
          alt="Skater Black Sweatshirt"
          class="sc-gsDKAQ"
        />
        <p
          class="sc-eCImPb hQahQF"
        >
          Skater Black Sweatshirt
        </p>
        <div
          class="sc-jRQBWg dlshIp"
        >
          <p
            class="sc-gKclnd jlmNbV"
          >
            <small>
              $
            </small>
            <b>
              25
            </b>
            <span>
              .90
            </span>
          </p>
          <p
            class="sc-iCfMLu kFBzSL"
          >
            <span>
              or 
              12
               x
            </span>
            <b>
              $
              2.16
            </b>
          </p>
        </div>
        <button
          class="sc-bdvvtL brAMjl"
          tabindex="-1"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
